src/node/cli.ts

```diff
@@ -240,6 +240,19 @@ export const optionDescriptions = (): string[] => {
   })
 }
 
+export function splitOnFirstEquals(str: string): string[] {
+  // we use regex instead of "=" to ensure we split at the first
+  // "=" and return the following substring with it
+  // important for the hashed-password which looks like this
+  // $argon2i$v=19$m=4096,t=3,p=1$0qR/o+0t00hsbJFQCKSfdQ$oFcM4rL6o+B7oxpuA4qlXubypbBPsf+8L531U7P9HYY
+  // 2 means return two items
+  // Source: https://stackoverflow.com/a/4607799/3015595
+  // We use the ? to say the the substr after the = is optional
+  const split = str.split(/=(.+)?/, 2)
```

Comment:
I am probably misunderstanding what you mean but I don't think that is what we want. (though I agree string splits are hard to read). 

We need the value to be split on the first equals and I don't think using a regex match would give us the same result (based on this quick test). 

![image](https://user-images.githubusercontent.com/3806031/121069259-20820580-c782-11eb-99c4-c6e238212513.png)

> If separator is a regular expression with capturing parentheses, then each time separator matches, the results (including any undefined results) of the capturing parentheses are spliced into the output array.

[MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#Capturing_parentheses)

Reply:
Ahhh, I see, I misunderstood what this code is doing. I guess it's because we wrote our own argument parser, and now need a way to parse: `--hashed-password=abc=def` into `key` = `hashed-password` and `value` = `abc=def`

Is there some reason we're implementing our own parser rather than [relying on a library like `yargs`](https://nodejs.org/en/knowledge/command-line/how-to-parse-command-line-arguments/)? That's too big of a change for this PR, but maybe we can consider adopting a library in the future.

Reply:
> Is there some reason we're implementing our own parser rather than relying on a library like yargs?

Fantastic question! All I can say is "legacy" 😛

But in all seriousness, I agree with you. Too big for this PR but something we want to do. I know @oxy has mentioned it multiple times. Hoping we get to it eventually!

Reply:
https://github.com/cdr/code-server/issues/1908